data/block/metaBlock_test.go

```diff
@@ -269,3 +270,61 @@ func TestMetaBlock_GetMiniBlockHeadersWithDst(t *testing.T) {
 	mbDst1 := metaHdr.GetMiniBlockHeadersWithDst(1)
 	assert.Equal(t, len(shardMBHeader), len(mbDst1))
 }
+
+func TestMetaBlock_GetOrderedCrossMiniblocksWithDstShouldWork(t *testing.T) {
+	t.Parallel()
+
+	metaHdr := &block.MetaBlock{Round: 6}
+	metaHdr.ShardInfo = make([]block.ShardData, 0)
+
+	shardMBHeader1 := make([]block.MiniBlockHeader, 0)
+	shMBHdr1 := block.MiniBlockHeader{SenderShardID: 0, ReceiverShardID: 1, Hash: []byte("hash1")}
+	shardMBHeader1 = append(shardMBHeader1, shMBHdr1)
+	shData1 := block.ShardData{Round: 11, ShardID: 0, HeaderHash: []byte("sh1"), ShardMiniBlockHeaders: shardMBHeader1}
+
+	shardMBHeader2 := make([]block.MiniBlockHeader, 0)
+	shMBHdr2 := block.MiniBlockHeader{SenderShardID: 0, ReceiverShardID: 1, Hash: []byte("hash2")}
+	shardMBHeader2 = append(shardMBHeader2, shMBHdr2)
+	shData2 := block.ShardData{Round: 9, ShardID: 0, HeaderHash: []byte("sh2"), ShardMiniBlockHeaders: shardMBHeader2}
+
+	shardMBHeader3 := make([]block.MiniBlockHeader, 0)
+	shMBHdr3 := block.MiniBlockHeader{SenderShardID: 2, ReceiverShardID: 1, Hash: []byte("hash3")}
+	shardMBHeader3 = append(shardMBHeader3, shMBHdr3)
+	shData3 := block.ShardData{Round: 10, ShardID: 2, HeaderHash: []byte("sh3"), ShardMiniBlockHeaders: shardMBHeader3}
+
+	shardMBHeader4 := make([]block.MiniBlockHeader, 0)
+	shMBHdr4 := block.MiniBlockHeader{SenderShardID: 2, ReceiverShardID: 1, Hash: []byte("hash4")}
+	shardMBHeader4 = append(shardMBHeader4, shMBHdr4)
+	shData4 := block.ShardData{Round: 8, ShardID: 2, HeaderHash: []byte("sh4"), ShardMiniBlockHeaders: shardMBHeader4}
+
+	shardMBHeader5 := make([]block.MiniBlockHeader, 0)
+	shMBHdr5 := block.MiniBlockHeader{SenderShardID: 1, ReceiverShardID: 2, Hash: []byte("hash5")}
+	shardMBHeader5 = append(shardMBHeader5, shMBHdr5)
+	shData5 := block.ShardData{Round: 7, ShardID: 1, HeaderHash: []byte("sh5"), ShardMiniBlockHeaders: shardMBHeader5}
+
+	metaHdr.ShardInfo = append(metaHdr.ShardInfo, shData1, shData2, shData3, shData4, shData5)
+
+	metaHdr.MiniBlockHeaders = append(metaHdr.MiniBlockHeaders, block.MiniBlockHeader{
```

Comment:
maybe add a miniblock with the destination == core.AllShardId

Reply:
Yep, I will add